pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -97,7 +97,19 @@ let
       };
     in stdenv.lib.makeOverridable drvScope (auto // manualArgs);
 
-  mkScope = scope: pkgs // pkgs.xorg // pkgs.gnome2 // { inherit stdenv; } // scope;
+  mkScope = scope: let
+      ps = pkgs.__splicedPackages;
+      scopeSpliced = pkgs.splicePackages {
+        pkgsBuildBuild = scope.buildHaskellPackages.buildHaskellPackages;
+        pkgsBuildHost = scope.buildHaskellPackages;
+        pkgsHostHost = {};
+        pkgsHostTarget = scope;
+        pkgsTargetTarget = {};
+      } // {
```

Comment:
Maybe add a parenthesis around ```pkgs.splicePackages {...}``` to make it more clear. I think that's how the precedence works but it is sometimes unclear whether this is ```(pkgs.splicePackages {...}) // {...}``` or ```pkgs.splicePackages ({...} // {...})```

Reply:
Maybe it's just all the Haskell, but I find function application before operators one of the few precedence I *do* remember :D.